cmd/minikube/cmd/start.go

```diff
@@ -312,3 +350,73 @@ func init() {
 	viper.BindPFlags(startCmd.Flags())
 	RootCmd.AddCommand(startCmd)
 }
+
+// saveConfig saves profile cluster configuration in
+// $MINIKUBE_HOME/profiles/<profilename>/config.json
+func saveConfig(clusterConfig cluster.Config) error {
+	data, err := json.MarshalIndent(clusterConfig, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	profilePath := filepath.Join(constants.GetMinipath(), "profiles", viper.GetString(cfg.MachineProfile))
+
+	if err := os.MkdirAll(profilePath, 0700); err != nil {
+		return err
+	}
+
+	if err := saveConfigToFile(data, filepath.Join(profilePath, "config.json")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func saveConfigToFile(data []byte, file string) error {
```

Comment:
As a future todo we should consolidate all these saveConfig and loadConfig functions.  As we have separate ones for our viper config, this config, and the docker machine config.